### ISSUE TITLE
Add a TFDS user CLI

### ISSUE THREAD

**Issue by Conchylicultor** [tensorflow/datasets#2161]
It would be nice to have a user CLI which is installed at the same time as TFDS to pre-generate dataset:

```sh
tfds new my_dataset  # Generate the base template & cie
cd my_dataset/
tfds build  # Generate my_dataset
```

`tfds new` would create the following folder structure:
```
my_dataset/
    fake_data/
    __init__.py
    dataset.py
    dataset_test.py
    make_fake_data.py
    checksum.txt
```

Some technical details:
* Script would be implemented in python using argparse
* Is added through the `entry_points={'command'}` section in `setup.py`

This first require to implement the one folder==one dataset model.

**Comment by acharles7** [comment]
can I work on this?

**Comment by vijayphoenix** [comment]
Hi @acharles7,
Thank you for your interest.

FYI, I am already working on the bigger part of this issue, i.e, `one folder per one dataset`.

**Comment by acharles7** [comment]
Thanks for letting me know, Let me know if you need any help:)

**Comment by Conchylicultor** [comment]
Thank you for your interest!

@acharles7, @vijayphoenix I think this bug and https://github.com/tensorflow/datasets/issues/2139 are complementary and both could be worked in parallel.

@acharles7, If you want to get started, you could start by splitting this task in small self-contained PRs:

create a new `scripts/cli/main.py` folder and add a dummy script.

* **PR 1**: Add a dummy `scripts/cli/main.py` script:
   * It should be registered in `setup.py` `entry_points`.
   * We want to use both `absl.app` with `argparse`. See https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/scripts/documentation/build_catalog.py#L150 for an example
   * The script could be executed, but wouldn't do anything

Then each new command could be added in a new PR:

* **PR 2**: Add the `tfds new <dataset>` command:
  * Each command should be implemented as a separate file, (here `scripts/cli/new.py`)
  * This would generate the `<dataset>/` folder structure
  * The implementation can be inspired by: https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/scripts/create_new_dataset.py
  * We want to use `pathlib.Pathlib` instead of `GFile`

**Comment by acharles7** [comment]
@Conchylicultor Thanks, working on it (PR1), I'll let you if I have any questions.

**Comment by acharles7** [comment]
Seems like @vijayphoenix already made architecture for CLI. Thanks
So, I'll add required new commands for this cli.

**Comment by acharles7** [comment]
@Conchylicultor @vijayphoenix Since we are implementing this CLI, When are we going to do refactoring on all datasets? (each dataset assets in one folder).

**Comment by Conchylicultor** [comment]
@acharles7,You're right. This is something we'll likely have to do at some point.

For this, we'll need some script to automate this refactoring. That's also something you can try if you're interested (write the refactoring script, not actually refactoring the datasets)

**Comment by acharles7** [comment]
Yes, I would love to work on this. 
So, It'll be a normal python script to aggregate all dataset assets in one folder. Right?

```bash
$ script.py dataset_name

# output:
dataset_name (a folder containing all files and folders(fake data, urlchecksum, test files etc.))
```

**Comment by Conchylicultor** [comment]
I don't think we want to run the script individually for each datasets. Instead, we would run the script once (`python refactor_dataset_to_folder.py` and it would refactor our entire database as folder (e.g. create all subfolder, rename imports,...)).

Note:
* The script don't have to work perfectly, and we would probably have many manual editions to do to fix edge cases.
* To detect which datasets to refactor, you can hardcode the folders `image/`, `text/`,... or auto-detect dataset with `tfds.list_builder`
* You can implement the script in multiple steps and send multiple PRs: 1st step: Refactor a single dataset, 2nd step: Refactor all datasets
* The tricky parts would be for the datasets which share utils (e.g. `wmt19.py`, `wmt18.py`,... which all use `wmt.py`). For now, we can omit those datasets (it's ok if the script don't work for those).
* Don't hesitate to send unfinished, draft PR to get early feedback. Let us know if you have questions.

**Comment by acharles7** [comment]
Sure, I'll start working on this step by step and let you know if I have any concerns.

**Comment by vijayphoenix** [comment]
For future work, see https://github.com/tensorflow/datasets/issues/2447
